### PR TITLE
nibhati-5779-add ResourceIdentifier

### PR DESCRIPTION
**Pull request by nisha-bhatia**
# Description
<i>Add your description here!</i>

# Checklist

To ensure a quick review and merge, please ensure:
- [ ] The PR has a understandable title and description explaining the _why_ and _what_.
- [ ] The PR is opened in draft if not ready for review yet.
   - If opened in draft, please allocate sufficient time (24 hours) after moving out of draft for review
- [x] The branch is recent enough to not have merge conflicts upon creation.

# Ready to Land?
- [ ] Build is completely green
   - Submissions with test failures require tracking issue and approval of a CODEOWNER
- [ ] At least one +1 review by a CODEOWNER
- [ ] All -1 reviews are confirmed resolved by the reviewer 
   - Override/Marking reviews stale must be discussed with CODEOWNERS first

### CODE REVIEW

**Comment by nisha-bhatia** [comment]
ADO item: https://dev.azure.com/azure-mgmt-ex/DotNET%20Management%20SDK/_boards/board/t/DotNET%20Management%20SDK%20Team/Stories/?workitem=5779

File: test/TestProjects/OperationGroupMappings/OperationGroupMappings.sln [hunk 1]
```diff
@@ -0,0 +1,25 @@
+﻿
```
Comment (m-nash) [inline test/TestProjects/OperationGroupMappings/OperationGroupMappings.sln:1]: remove this file

**Review by m-nash — CHANGES_REQUESTED** [review]

**Comment by chamons** [comment]
The tl;dr diff - https://gist.github.com/ea72d5c53c8cde4b388c38dc91b07533

**Review by m-nash — APPROVED** [review]

File: src/AutoRest.CSharp/Common/Output/Models/Types/SystemObjectType.cs [hunk 2]
```diff
@@ -17,19 +17,18 @@ namespace AutoRest.CSharp.Output.Models.Types
...
         {
             _type = type;
-            _csharpType = new CSharpType(type);
         }
 
         public override ObjectTypeProperty? AdditionalPropertiesProperty => null;
 
         protected override string DefaultName => GetNameWithoutGeneric(_type);
         protected override string DefaultAccessibility { get; } = "public";
+        protected override string DefaultNamespace => _type.Namespace ?? base.DefaultNamespace;
```
Comment (pakrym) [inline src/AutoRest.CSharp/Common/Output/Models/Types/SystemObjectType.cs:31]: is `_type.Namespace` ever null?

**Comment by nisha-bhatia** [comment]
_type.Namespace can be null, as the type for Namespace in the Type class is string?

**Comment by pakrym** [comment]
I don't think the base.DefaultNamespace would be the right thing to use in that case anyway.

**Pull request merged**